config/cmake/modules/FindMAGMA.cmake

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) 2010-2024, Lawrence Livermore National Security, LLC. Produced
+# at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+# LICENSE and NOTICE for details. LLNL-CODE-806117.
+#
+# This file is part of the MFEM library. For more information and source code
+# availability visit https://mfem.org.
+#
+# MFEM is free software; you can redistribute it and/or modify it under the
+# terms of the BSD-3 license. We welcome feedback and contributions, see file
+# CONTRIBUTING.md for details.
+
+# Defines the following variables:
+#   - MAGMA_FOUND
+#   - MAGMA_LIBRARIES
+#   - MAGMA_INCLUDE_DIRS
+
+include(MfemCmakeUtilities)
+set(MAGMA_REQUIRED_LIBRARIES cublas cusparse)
```

Comment:
These GPU libraries should be added after detecting how MAGMA was built -- with CUDA or HIP. See `FindHYPRE.cmake` for an example how this can be done.

Reply:
We probably just want to error out if MFEM and MAGMA are not either both built with CUDA or both built with HIP

Reply:
Yes, but first we need to detect how MAGMA was built. And for that one can look at `FindHYPRE.cmake` for an example how that can be done.

Reply:
Auto-detecting CUDA or HIP in MAGMA is not absolutely necessary but it will be nice to have -- up to you.

Regardless of that, the line
```cmake
set(MAGMA_REQUIRED_LIBRARIES cublas cusparse)
```
needs to be removed because it will overwrite the value set in `defaults.cmake`. In general, if needed, we can append to the list here, instead of overwriting it.